parasol/solr/client.py

```diff
@@ -0,0 +1,131 @@
+import inspect
+import json
+import logging
+import sys
+import time
+from urllib.parse import urljoin
+
+from attrdict import AttrDict
+import requests
+
+# Start making Django agnostic from the get-go
+try:
+    from django.settings import settings
+    from django.core.exceptions import ImproperlyConfigured
+except ImportError:
+    pass
+
+
+from parasol import __version__ as parasol_version
+from parasol.solr.schema import Schema
+from parasol.solr.update import Update
+from parasol.solr.admin import CoreAdmin
+
+logger = logging.getLogger(__name__)
+
+
+
+class SolrClientException(Exception):
+    '''Base class for all exceptions in this module'''
+    pass
+
+
+class CoreForTestExists(SolrClientException):
+    '''Raised when default core for running unit tests exists'''
+
+
+
+class SolrClient:
+    '''Base class for all SolrClient with sane development defaults'''
+    #: Url for solr base instance
+    solr_url = 'http://localhost:8983/solr'
```

Comment:
This seems like a weird default to me. I'd rather see url be a required option.

Reply:
Done (re: comments below).